lua/autorun/server/sv_downloader_init.lua

```diff
@@ -24,13 +27,46 @@ local context = {
     usingAddons = {},
     started = SysTime(),
     gamemodeAddons = {},
-    addonsToCache = {}
+    addonsToCache = {},
+    manualAddons = {},
+    scanResult = {}
 }
 
 if not file.Exists(context.dataFolder, "DATA") then
     file.CreateDir(context.dataFolder)
 end
 
+net.Receive("uwd_exchange_scan_result", function(len, ply)
+    local scanResult = table.Copy(context.scanResult)
+
+    for wsid, value in pairs(context.manualAddons) do
+        if scanResult[wsid] then
+            scanResult[wsid].cachedManual = value
+        end
+    end
+
+    scanResult = util.TableToJSON(scanResult)
+
+    net.Start("uwd_exchange_scan_result")
+    net.WriteString(scanResult) -- Lazy
```

Comment:
I wonder if we should compress this before sending on the networking, it can be a huge string (max is 65532 characters)

Reply:
lol that's why it's "lazy", I'm not properly sending the data yet.

Reply:
There it is, it's not lazy anymore: e3ac7c6d9a48b7f2527db703640bd81b9f7910a9
Now we compress the table and send the binary result in chunks. It scales whenever we need it.